libdnf/repo/solvable/Dependency.cpp

```diff
@@ -0,0 +1,157 @@
+#include "Dependency.hpp"
+#include "libdnf/utils/utils.hpp"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+#   include <solv/pool_parserpmrichdep.h>
+#ifdef __cplusplus
+}
+#endif
+
+
+#define NUMBER_OF_ITEMS_IN_REL_DEPENDENCY 3
+
+class Dependency::Impl {
+public:
+    Impl();
+    Impl(const Impl &source);
+    ~Impl() = default;
+private:
+    friend class Dependency;
+
+    void parseAndCreateDependency(const std::string &dependency);
+    void createRelationalDependency(const std::vector<std::string> &results);
+    bool isRichDependency(const std::string &dependency) const;
+    int determineComparisonType(const std::string &type);
+    Pool *getPool();
+
+    Id libsolvableId;
```

Comment:
Is there any Id that is not part of libsolv? I guess that no, therefore name can be shorter.